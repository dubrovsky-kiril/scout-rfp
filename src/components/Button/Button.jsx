import React, { memo } from "react";
import PropTypes from "prop-types";
import styles from "./Button.scss";

const Button = ({ children, isDisabled, onClick }) => {
  return (
    <button
      className={styles.button}
      disabled={isDisabled}
      onClick={onClick}
      type="button"
    >
      {children}
    </button>
  );
};

Button.defaultProps = {
  isDisabled: false
};

Button.propTypes = {
  children: PropTypes.string.isRequired,
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

export default memo(Button);
