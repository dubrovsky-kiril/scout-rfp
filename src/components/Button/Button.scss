.button {
  background-color: palevioletred;
  color: #fff;
  font-size: 18px;
  padding: 15px 30px;
  outline: none;
  font-weight: bolder;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: 0.3s;
  width: 100%;
  border: none;
  margin-top: 30px;
  &:hover {
    opacity: 0.8;
    transition: 0.3s;
  }
  &:active {
    opacity: 0.9;
    transition: 0.3s;
  }
}
