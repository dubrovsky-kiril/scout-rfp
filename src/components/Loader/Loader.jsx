import React from "react";
import styles from "./Loader.scss";

const Loader = () => {
  return (
    <div className={styles.lodaer_container}>
      <div className={styles.loader}>
        <div />
        <div />
      </div>
    </div>
  );
};

export default Loader;
