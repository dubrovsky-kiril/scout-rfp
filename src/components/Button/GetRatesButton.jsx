import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { fetchLatestRates } from "#store/actions";
import Button from "#components/Button/Button";

const mapStateToProps = state => {
  const { isFetching } = state;

  return {
    isDataFetching: isFetching
  };
};

const GetRatesButton = ({ dispatch, isDataFetching }) => {
  return (
    <Button
      isDisabled={isDataFetching}
      onClick={() => dispatch(fetchLatestRates())}
    >
      Get rates
    </Button>
  );
};

GetRatesButton.propTypes = {
  isDataFetching: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired
};

export default connect(mapStateToProps)(React.memo(GetRatesButton));
