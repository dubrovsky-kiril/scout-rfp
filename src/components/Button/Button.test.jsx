import React from "react";
import { create } from "react-test-renderer";
import { mount, shallow } from "enzyme";
import Loader from "#components/Loader/Loader";
import Button from "./Button";

describe("Button component", () => {
  test("Matches the snapshot", () => {
    const button = create(<Button onClick={() => {}}>Submit</Button>);

    expect(button.toJSON()).toMatchSnapshot();
  });
  test("Is clickable", () => {
    const mockCallBack = jest.fn();

    const button = shallow(<Button onClick={mockCallBack}>Submit</Button>);

    button.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
  test("Is unclickable when disabled", () => {
    const mockCallBack = jest.fn();

    const button = mount(
      <Button isDisabled onClick={mockCallBack}>
        Submit
      </Button>
    );

    button.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(0);
  });
  test("Is displaying exptected text", () => {
    const button = shallow(<Button onClick={() => {}}>Submit</Button>);

    expect(button.text()).toEqual("Submit");
  });
  test("Is displaying Loader component when pending", () => {
    const button = shallow(
      <Button isPending onClick={() => {}}>
        Submit
      </Button>
    );

    expect(button.contains(<Loader />)).toBe(true);
  });
});
