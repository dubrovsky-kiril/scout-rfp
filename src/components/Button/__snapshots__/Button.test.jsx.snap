// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Submit
</button>
`;
